cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamFetchHandler.java

```diff
@@ -121,6 +129,8 @@ public void fetch(HttpRequest request, final HttpResponder responder,
       return;
     }
 
+    // Make sure the user has READ permission on the stream since getConfig doesn't check for the same.
+    authorizationEnforcer.enforce(streamId.toEntityId(), authenticationContext.getPrincipal(), Action.READ);
```

Comment:
Well, there is really no one thing at the layer below. If you see the code, majority of the logic is in the handler implementation itself. It will be a lot of work to push it down which I feel is not really necessary.